fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountAssembler.java

```diff
@@ -323,10 +324,10 @@ public SavingsAccount assembleFrom(final JsonCommand command, final AppUser subm
         }
 
         final SavingsAccount account = SavingsAccount.createNewApplicationForSubmittal(client, group, product, fieldOfficer, accountNo,
-                externalId, accountType, submittedOnDate, submittedBy, interestRate, interestCompoundingPeriodType,
-                interestPostingPeriodType, interestCalculationType, interestCalculationDaysInYearType, minRequiredOpeningBalance,
-                lockinPeriodFrequency, lockinPeriodFrequencyType, iswithdrawalFeeApplicableForTransfer, charges, allowOverdraft,
-                overdraftLimit, enforceMinRequiredBalance, minRequiredBalance, maxAllowedLienLimit, lienAllowed,
+                ExternalIdFactory.produce(externalId), accountType, submittedOnDate, submittedBy, interestRate,
```

Comment:
Here we should generate one if nothing was provided: `ExternalIdFactory.create(...)`